virtual_rainforest/models/abiotic/energy_balance.py

```diff
@@ -0,0 +1,321 @@
+    0.32  # typical for decidious forest with wind above canopy 2 m/s
+)
+AIR_CONDUCTIVITY = 50.0  # typical for decidious forest with wind above canopy 2 m/s
+MIN_LEAF_AIR_CONDUCTIVITY = (
+    0.13  # typical for decidious forest with wind above canopy 2 m/s
+)
+MAX_LEAF_AIR_CONDUCTIVITY = (
+    0.19  # typical for decidious forest with wind above canopy 2 m/s
+)
+KARMANS_CONSTANT = 0.4  # constant to calculate mixing length
+CELCIUS_TO_KELVIN = 273.15  # calculate absolute temperature in Kelvin
+STANDARD_MOLE = 44.6  # moles of ideal gas in 1 m^3 air at standard atmosphere
+MOLAR_HEAT_CAPACITY_AIR = 29.19  # molar heat capacity of air [J mol-1 C-1]
+VAPOR_PRESSURE_FACTOR1 = 0.6108  # constant in calculation of vapor pressure
+VAPOR_PRESSURE_FACTOR2 = 17.27  # constant in calculation of vapor pressure
+VAPOR_PRESSURE_FACTOR3 = 237.7  # constant in calculation of vapor pressure
+
+# import external driving data at reference hight (2 m) for current timestep
+# this will be a Data.data object
+data = {
+    "air_temperature_2m": 25,
+    "relative_humidity_2m": 90,
+    "atmospheric_pressure_2m": 101.325,
+    "wind_speed_10m": 2.0,
+    "soil_moisture_reference": 0.3,
+}
+
+# soil parameter for different soil types, no real values, will be external data base
+soil_parameters = {  # these are not real values
+    "Sand": {
+        "Smax": 1,  # Volumetric water content at saturation (m^3 / m^3)
+        "Smin": 1,  # Residual water content (m^3 / m^3)
+        "alpha": 1,  # Shape parameter of the van Genuchten model (cm^-1)
+        "n": 1,  # Pore_size_distribution_parameter (dimensionless, > 1)
+        "Ksat": 1,  # Saturated hydraulic conductivity (cm / day)
+        "Vq": 1,  # Volumetric quartz content of soil
+        "Vm": 1,  # Volumetric mineral content of soil
+        "Vo": 1,  # Volumetric organic content of soil
+        "Mc": 1,  # Mass fraction of clay
+        "rho": 1,  # Soil bulk density (Mg / m^3)
+        "b": 1,  # Shape parameter for Campbell model (dimensionless, > 1)
+        "psi_e": 1,  # Matric potential (J / m^3)
+    },
+    "Clay": {
+        "Smax": 0.1,  # Volumetric water content at saturation (m^3 / m^3)
+        "Smin": 0.1,  # Residual water content (m^3 / m^3)
+        "alpha": 0.1,  # Shape parameter of the van Genuchten model (cm^-1)
+        "n": 0.1,  # Pore_size_distribution_parameter (dimensionless, > 1)
+        "Ksat": 0.1,  # Saturated hydraulic conductivity (cm / day)
+        "Vq": 0.1,  # Volumetric quartz content of soil
+        "Vm": 0.1,  # Volumetric mineral content of soil
+        "Vo": 0.1,  # Volumetric organic content of soil
+        "Mc": 0.1,  # Mass fraction of clay
+        "rho": 0.1,  # Soil bulk density (Mg / m^3)
+        "b": 0.1,  # Shape parameter for Campbell model (dimensionless, > 1)
+        "psi_e": 0.1,  # Matric potential (J / m^3)
+    },
+}
+
+
+class EnergyBalance:
+    """EnergyBalance method.
+
+    Attributes:
+        air_temperature: NDArray[np.float32],
```

Comment:
Don't know if giving type information here is necessary, as it's already covered by the init function signature

Reply:
Also think that description of what the attributes are would be useful here

Reply:
I'd personally *not* include the attribute names here, though it isn't exactly wrong and I haven't checked what style you guys are using elsewhere.

Instead, just add a docstring after the attributes when you create them in `__init__()`, e.g.:

```py
self.my_attr = 5.0
"""A very important parameter between 0.0 and 10.0"""
```

Again, this way your IDE and documentation generator can pick this up.